.github/workflows/ci.yml

```diff
@@ -1,45 +1,52 @@
 name: Continuous Integration
 
 on:
-  push:
-  pull_request:
+  pull_request_target:
+    branches: [ master ]
 
 jobs:
   htmlproofer:
     name: HTMLProofer
+
     runs-on: ubuntu-latest
-    
+
     steps:
       - name: Checkout EIP Repository
         uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
-      
+        with:
+          ref: "refs/pull/${{ github.event.number }}/merge"
+
       - name: Install OpenSSL
         run: sudo apt-get update && sudo apt-get install -y libcurl4-openssl-dev
-    
+
       - name: Install Ruby
         uses: ruby/setup-ruby@08245253a76fa4d1e459b7809579c62bd9eb718a
         with:
-          ruby-version: '2.6.0'
+          ruby-version: 2.6.0
           bundler-cache: true
-        
+
       - name: Build Website
         run: |
           bundle exec jekyll doctor
           bundle exec jekyll build
-      
+
       - name: HTML Proofer
         run: bundle exec htmlproofer ./_site --check-html --check-opengraph --report-missing-names --log-level=:debug --assume-extension --empty-alt-ignore --timeframe=6w --disable-external
-      
+
       - name: DNS Validator
         run: bundle exec github-pages health-check
   
   codespell:
     name: CodeSpell
+
     runs-on: ubuntu-latest
+
     steps:
       - name: Checkout EIP Repository
         uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
-        
+        with:
+          ref: "refs/pull/${{ github.event.number }}/merge"
```

Comment:
What does this do?

Reply:
Pulls the files from the PR. I changed the event to `pull_request_target`, which always runs (even for new contributors) but also changes the default checkout behaviour to checkout the master branch of the original repo.